petab/v1/priors.py

```diff
@@ -58,6 +58,15 @@ class Prior:
         on the `parameter_scale` scale).
     :param bounds: The untransformed bounds of the sample (lower, upper).
     :param transformation: The transformation of the distribution.
+    :param bounds_truncate: Whether the generated prior will be truncated
+        at the bounds.
+        If ``True``, the probability density will be rescaled
+        accordingly and the sample is generated from the truncated
+        distribution.
+        If ``False``, the probability density will not account for the
+        bounds, but any parameter samples outside the bounds will be set to
+        the value of the closest bound. In this case, the PDF might not match
+        the sample.
```

Comment:
True: new behavior
False: old behavior

PEtab specs are ambiguous there (https://github.com/PEtab-dev/PEtab/blob/b9e141dd75798d179c17262f085ed6cef8555b3e/doc/v1/documentation_data_format.rst?plain=1#L527-L529):

> Sampled points are clipped to lie inside the parameter boundaries specified by ``lowerBound`` and ``upperBound``.

While I think the new behavior is more correct, I will wait another while before merging this.

Reply:
> PEtab specs are ambiguous there (https://github.com/PEtab-dev/PEtab/blob/b9e141dd75798d179c17262f085ed6cef8555b3e/doc/v1/documentation_data_format.rst?plain=1#L527-L529):
> 
> > Sampled points are clipped to lie inside the parameter boundaries specified by `lowerBound` and `upperBound`.
> 
> While I think the new behavior is more correct, I will wait another while before merging this.

I agree, but I would also be in favor of removing the old behavior entirely. Or "fix" it by resampling out-of-bounds samples.

Reply:
Agreed that we should get rid of that. Happy to remove this option completely.

I will wait for some feedback to https://github.com/PEtab-dev/PEtab/issues/591 before proceeding.